_source/_posts/2021-09-30-oauth-sessions-with-java.adoc

```diff
@@ -0,0 +1,316 @@
+
+[source,sh]
+----
+docker compose up
+----
+
+NOTE: You can stop the process by pressing `Ctrl`+`C` or running the `docker compose down` command from the same directory.
+
+== Build a Secure Spring Boot Application
+
+Now that the system dependencies are out of the way, let's move on to building a Spring Boot application.
+
+Create a new Spring Boot application by visiting https://start.spring.io[`start.spring.io`] and selecting the `Web` and `Okta` dependencies or by running the following command:
+
+[source,sh]
+----
+http https://start.spring.io/starter.tgz \
+  bootVersion==2.5.4 \
+  dependencies==web,okta \
+  groupId==com.example \
+  artifactId==webapp \
+  name=="Web Application" \
+  description=="Demo Web Application" \
+  packageName==com.example \
+  javaVersion==11 \
+| tar -xzvf -
+----
+
+To give a visual indication of which server handled the request, create a REST controller that displays the server port: `src/main/java/com/example/Endpoints.java`:
+
+[source,java]
+----
+package com.example;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class Endpoints {
+
+    @GetMapping("/")
+    String serverInfo(@Value("${server.port}") int port) {
+        return "Hello, your server port is: " + port;
+    }
+}
+----
+
+=== Start the Spring Boot Application
+
+In the previous section, HAproxy was bound to port `8080`, which is also the default port for Spring Boot; start the application on port `8081` using:
+
+[source,sh]
+----
+SERVER_PORT=8081 ./mvnw spring-boot:run
+----
+
+At this point, the Spring Boot application has NOT been configured to use Redis or OAuth, but you can still test out the server by grabbing the auto-generated password from the console output; it will look something like this:
+
+[source,txt]
+----
+Using generated security password: 4302a714-580b-4d01-91d9-5d9597ee1bb5
+----
+
+Copy the password and make a request to the Spring application:
+
+[source,sh]
+----
+http :8081/ --auth user:<your-password>
+----
+
+You will see a response containing the server port:
+
+[source,txt]
+----
+Hello, your server port is: 8081
+----
+
+Great, this means that the Spring Boot application is up and running! Now make sure you can access the server through the load balancer on port `8080`:
+
+[source,sh]
+----
+http :8080/ --auth user:<your-password>
+----
+
+You should see the same response; if you see a `503 Service Unavailable`, try the request one more time.
+
+IMPORTANT: The HAproxy configuration used in this post does NOT have health checks enabled, so it blindly will alternate requests between port `8081` and `8082`; this is intentional to simplify the configuration. Take a look at https://docs.spring.io/spring-boot/docs/current/reference/html/actuator.html[Spring Actuator] if you want to add health checks and other monitoring to your application.
```

Comment:
```suggestion
IMPORTANT: The HAproxy configuration used in this post does NOT have health checks enabled, so it blindly will alternate requests between port `8081` and `8082`; this is intentional to simplify the configuration. Take a look at https://docs.spring.io/spring-boot/docs/current/reference/html/actuator.html[Spring Actuator] if you want to add health checks and other monitoring to your application.
```
```suggestion
IMPORTANT: The HAproxy configuration used in this post does NOT have health checks enabled, so it will alternate requests between port `8081` and `8082`; this is intentional to simplify the configuration. Take a look at https://docs.spring.io/spring-boot/docs/current/reference/html/actuator.html[Spring Actuator] if you want to add health checks and other monitoring to your application.
```
removed 'blindly' since it feels unnecessary here, and possibly 'ablist' - if you need an adjective, could 'randomly' do the job?